node/network/bridge/src/tx/mod.rs

```diff
@@ -148,14 +150,31 @@ where
 	AD: validator_discovery::AuthorityDiscovery + Clone,
 {
 	match msg {
-		NetworkBridgeTxMessage::ReportPeer(peer, rep) => {
-			if !rep.is_benefit() {
+		NetworkBridgeTxMessage::ReportPeer(ReportPeerMessage::Single(peer, rep)) => {
+			if !rep.value.is_positive() {
 				gum::debug!(target: LOG_TARGET, ?peer, ?rep, action = "ReportPeer");
 			}
 
 			metrics.on_report_event();
 			network_service.report_peer(peer, rep);
 		},
+		NetworkBridgeTxMessage::ReportPeer(ReportPeerMessage::Batch(batch)) => {
+			let reports: Vec<(PeerId, ReputationChange)> = batch
```

Comment:
Why do we need this intermediate `Vec` if we can just iterate over the original hash table?